code/game/machinery/atmoalter/clamp.dm

```diff
@@ -80,39 +69,31 @@
 
 	qdel(target.parent)
 
-	if(network_node1)
-		qdel(network_node1)
-	if(network_node2)
-		qdel(network_node2)
-
-	var/obj/machinery/atmospherics/pipe/node1 = null
-	var/obj/machinery/atmospherics/pipe/node2 = null
+	for(var/datum/pipe_network/netnode in network_nodes)
+		qdel(netnode)
+	network_nodes.Cut()
 
-	if(target.node1)
-		target.node1.build_network()
-		node1 = target.node1
-	if(target.node2)
-		target.node2.build_network()
-		node2 = target.node2
-	if(istype(node1) && node1.parent)
-		var/datum/pipeline/P1 = node1.parent
-		P1.build_pipeline(node1)
-		qdel(P1)
-	if(istype(node2) && node2.parent)
-		var/datum/pipeline/P2 = node2.parent
-		P2.build_pipeline(node2)
-		qdel(P2)
-//  P1.build_network()
-//  P2.build_network()
+	for(var/obj/machinery/atmospherics/pipe/node in target.pipeline_expansion())
+		if(!istype(node))
```

Comment:
FYI this is extraneous since a `for(var/path/foo in var)` loop will only find things of that part. NBD though since this is a small list and not called often